ddpui/ddpdbt/dbt_service.py

```
@@ -937,3 +959,246 @@ def cleanup_unused_sources(org: Org, orgdbt: OrgDbt, manifest_json=None):
+
+    # Clean existing references
+    try:
+        if dbt_project_dir.exists():
+            shutil.rmtree(dbt_project_dir)
+            logger.info(f"Removed existing dbt directory: {dbt_project_dir}")
+
+        # remove canvas nodes and edges related to the dbt project
+        CanvasNode.objects.filter(orgdbt=orgdbt).delete()
+
+    except Exception as e:
+        logger.error(f"Failed to clean up existing directory: {e}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)